### ISSUE TITLE
Accessibility issue for Modal and fix FocusLock

### ISSUE THREAD

**Issue by JessicaMarine** [ustaxcourt/ef-cms#7819]
This is borrowed from the Flexion refactoring backlog. We could possibly call on 18F phone-a-friend to assist. The issue was surrounding hitting escape while a modal is open. 

> Title: Fix FocusLock component this.restoreFocus()
> Restoring focus back to the main page when modals close was causing problems when exiting Add Practitioner modals and scrolling to the success notification at the top.

From Andrew Dunkman (18F):
Asked around a bit, and the consensus is that focus should be returned to the item which spawned the modal (for example, the button clicked which caused the modal to appear).

I also got a plug for the new USWDS modal component (https://designsystem.digital.gov/components/modal/) which implements that focus flow.

**Comment by kkoskelin** [comment]
@adunkman in one particular workflow, that of adding a practitioner to a case, the button which spawned the modal could receive focus after the modal is dismissed, however, there is a success message which probably ought to receive focus.  Would you agree?  Happy to demo this for you.

**Comment by kkoskelin** [comment]
https://github.com/ustaxcourt/ef-cms/pull/1072

**Comment by adunkman** [comment]
Just summarizing our chat from yesterday — thanks for the demo! 

- If possible, focus the element which spawned the modal. 
- In cases where the element is no longer present (like, when removing people from cases, so the delete button is no longer there) the main content element should be focused (that is, the item which is focused when hitting "skip to main content"

I think that’s where we landed — give me a shout if I captured that wrong!

**Comment by kkoskelin** [comment]
@adunkman I have satisfied the first of the two requests.  As for the second, I do not know that such a requirement is directly possible, because it maybe only AFTER the modal has been removed from the DOM- (and it's deconstruction hooks fired) that the then-focused element is removed.
I can imagine attempting to fire a focus-related handler on a timeout (say, 250ms), but am unsure of the consequences.  The default behavior of MacOS VO seems to be to place the focus on the "skip to content" link already, which puts VO users just one link away from your request.  Thoughts?

**Comment by adunkman** [comment]
>As for the second…

I feel like this is what you were trying to explain to me during our call but I wasn’t quite getting it. Thanks for writing it out! 

>I can imagine attempting to fire a focus-related handler on a timeout (say, 250ms), but am unsure of the consequences.

Agreed, that’s a bit janky and feels like it would come back to bite us later on. 

>The default behavior of MacOS VO seems to be to place the focus on the "skip to content" link already, which puts VO users just one link away from your request. Thoughts?

Sounds like a good compromise! I’m happy with it! :gavel:

**Comment by adunkman** [comment]
This looks great! Confirmed on `test`.

https://user-images.githubusercontent.com/14930/115075099-95df0180-9ec8-11eb-8e53-1f3d141bc526.mov
